crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -423,6 +422,11 @@ impl SlidingSync {
             }
         }
 
+        // Apply the transaction id if one was generated.
+        if let Some(txn_id) = txn_id.as_ref() {
```

Comment:
`as_ref` should not be necessary:

```suggestion
        if let Some(txn_id) = txn_id {
```